2.rb

```diff
@@ -2,47 +2,64 @@
 
 require 'json'
 
-class IndexedDb
-  @file_name = 'indexedDb.txt'
-  @index_name = 'index.json'
-  @delim = ';'
-  @hash = JSON.parse(File.read(@index_name)) || {}
+class PersistedHash
+  attr_reader :file_name, :hash
 
-  def self.db_set(key, val)
-    index = File.size(@file_name)
-    @hash[key.to_sym] = index
-    File.write(@file_name, "#{key}:#{val}#{@delim}", mode: 'a')
-    IndexedDb.save_hash
+  def initialize(file_name = 'index.json')
+    @file_name = file_name
+    @hash = JSON.load_file(file_name) || {}
   end
 
-  def self.db_get(key)
-    output = nil
+  def [](key)
+    hash[key]
+  end
+
+  def []=(key, byte_offset)
+    hash[key] = byte_offset
+    save_hash
+  end
 
-    case key
-    when String
-      output = IndexedDb.db_get_from_index(@hash[key])
-    when Symbol
-      output = IndexedDb.db_get_from_index(@hash[key.to_s])
+  private
+
+  def save_hash
+    File.open(file_name, 'w+') do |f|
+      f << hash.to_json
     end
-    output
   end
+end
+
+class IndexedDb
+  DELIMINATOR = ';'
```

Comment:
If you don't intend to change this value, a constant is more idiomatic - as opposed to file_name, which I think could be something a user can configure.